.pre-commit-config.yaml

```diff
@@ -0,0 +1,65 @@
+# pre-commit is a tool to perform a predefined set of tasks manually and/or
+# automatically before git commits are made.
+#
+# Config reference: https://pre-commit.com/#pre-commit-configyaml---top-level
+#
+# Common tasks
+#
+# - Run on all files:   pre-commit run --all-files
+# - Register git hooks: pre-commit install --install-hooks
+#
+repos:
+  # Autoformat: Python code, syntax patterns are modernized
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v3.19.1
+    hooks:
+      - id: pyupgrade
+        args:
+          - --py37-plus
```

Comment:
what version of python do we require here? since we deploy our own image, this can be high! I recommend latest (3.13) and make the docker image match

Reply:
@yuvipanda I've changed this to 3.12 for now to match the Python version of the Ubuntu 24.04 image.